Spanner/tests/System/TransactionTest.php

```diff
@@ -264,6 +265,144 @@ public function testRunTransactionWithDbRole($db, $values, $expected)
         }
     }
 
+    /**
+     * @dataProvider getDirectedReadOptions
+     */
+    public function testTransactionExecuteWithDirectedRead($directedReadOptions)
+    {
+        $db = self::$database;
+        $snapshot = $db->snapshot();
+        $expected = 'Directed reads can only be performed in a read-only transaction.';
+        $exception = null;
+
+        try {
+            $row = $db->execute(
+                'SELECT * FROM ' . self::$tableName,
+                ['transactionId' => $snapshot->id()] + $directedReadOptions
+            )->rows()->current();
+        } catch (ServiceException $e) {
+            $exception = $e;
+        }
+        $this->assertNull($exception);
```

Comment:
instead of using a try..catch and asserting on a null exception, assert on row count instead

Reply:
Done